### ISSUE TITLE
Uncaught TypeError: Cannot read property 'end' of undefined

### ISSUE THREAD

**Issue by keyi** [sveale/remote-edit#118]
[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 1.0.7
**System**: Mac OS X 10.10.5
**Thrown From**: [remote-edit](https://github.com/sveale/remote-edit) package, v1.8.15

### Stack Trace

Uncaught TypeError: Cannot read property 'end' of undefined

```
At /Users/Ke/.atom/packages/remote-edit/lib/model/sftp-host.coffee:155

TypeError: Cannot read property 'end' of undefined
  at /Users/Ke/.atom/packages/remote-edit/lib/model/sftp-host.coffee:155:13
  at /Applications/Atom.app/Contents/Resources/app.asar/node_modules/request/node_modules/form-data/node_modules/async/lib/async.js:541:30
  at /Users/Ke/.atom/packages/remote-edit/node_modules/ssh2/lib/client.js:728:14
  at SSH2Stream.<anonymous> (/Users/Ke/.atom/packages/remote-edit/node_modules/ssh2/lib/client.js:1039:5)
  at SSH2Stream.g (events.js:257:16)
  at emitOne (events.js:77:13)
  at SSH2Stream.emit (events.js:166:7)
  at parsePacket (/Users/Ke/.atom/packages/remote-edit/node_modules/ssh2/node_modules/ssh2-streams/lib/ssh.js:3428:10)
  at SSH2Stream._transform (/Users/Ke/.atom/packages/remote-edit/node_modules/ssh2/node_modules/ssh2-streams/lib/ssh.js:555:13)
  at SSH2Stream.Transform._read [as __read] (_stream_transform.js:167:10)
  at SSH2Stream._read (/Users/Ke/.atom/packages/remote-edit/node_modules/ssh2/node_modules/ssh2-streams/lib/ssh.js:213:15)
  at SSH2Stream.Transform._write (_stream_transform.js:155:12)
  at doWrite (_stream_writable.js:295:12)
  at writeOrBuffer (_stream_writable.js:282:5)
  at SSH2Stream.Writable.write (_stream_writable.js:210:11)
  at Socket.ondata (_stream_readable.js:524:20)
  at emitOne (events.js:77:13)
  at Socket.emit (events.js:166:7)
  at readableAddChunk (_stream_readable.js:146:16)
  at Socket.Readable.push (_stream_readable.js:109:10)
  at TCP.onread (net.js:517:20)

```

### Commands

```
 14x -0:41.2.0 core:move-down (atom-text-editor.editor.mini.is-focused)
  3x -0:30.9.0 core:confirm (atom-text-editor.editor.mini.is-focused)
  3x -0:24.7.0 core:move-down (atom-text-editor.editor.mini.is-focused)
  2x -0:23.9.0 core:move-up (atom-text-editor.editor.mini.is-focused)
     -0:23 core:move-down (atom-text-editor.editor.mini.is-focused)
     -0:22.7.0 core:confirm (atom-text-editor.editor.mini.is-focused)
  9x -0:22 core:move-down (atom-text-editor.editor.mini.is-focused)
     -0:19.4.0 core:confirm (atom-text-editor.editor.mini.is-focused)
  7x -0:18.7.0 core:move-down (atom-text-editor.editor.mini.is-focused)
     -0:13.6.0 core:confirm (atom-text-editor.editor.mini.is-focused)
  4x -0:12.7.0 core:move-down (atom-text-editor.editor.mini.is-focused)
     -0:11.1.0 core:confirm (atom-text-editor.editor.mini.is-focused)
  6x -0:10.4.0 core:move-down (atom-text-editor.editor.mini.is-focused)
  2x -0:08.3.0 core:move-up (atom-text-editor.editor.mini.is-focused)
  3x -0:02.9.0 core:move-down (atom-text-editor.editor.mini.is-focused)
     -0:01.4.0 core:confirm (atom-text-editor.editor.mini.is-focused)
```

### Config

```json
{
  "core": {
    "themes": [
      "atom-material-ui",
      "atom-material-syntax"
    ]
  },
  "remote-edit": {}
}
```

### Installed Packages

```coffee
# User
atom-beautify, v0.28.11
atom-material-syntax, v0.2.3
atom-material-ui, v0.6.0
autocomplete-paths, v1.0.2
color-picker, v2.0.11
emmet, v2.3.12
file-icons, v1.6.7
javascript-snippets, v1.0.0
linter, v1.4.3
linter-jshint, v1.1.5
minimap, v4.13.1
pigments, v0.11.0
project-manager, v1.15.11
remote-edit, v1.8.15
vim-mode, v0.58.0

# Dev
No dev packages
```

**Comment by ruvasik** [comment]
+1
after last update, all hosts have been removed and now new connections fail

**Comment by fojt-cz** [comment]
try restart Atom

**Comment by ruvasik** [comment]
many times:)
and macos even

**Comment by sveale** [comment]
I've just published a new version, 1.8.19, give it a whirl and let me know if it resolves the issue.
